File: output/xCodeEval_remove_8000/Java/Go/c60f78608a1a11e7f7f98789f3e9931c.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

type Point struct {
	X int
	Y int
}

type Edge struct {
	Vertex0       *Vertex
	Vertex1       *Vertex
	Bidirectional bool
}

type Vertex struct {
	Index int
	Edges []*Edge
}

type Set []*Point

func (s Set) Len() int {
	return len(s)
}

func (s Set) Less(i, j int) bool {
	if s[i].X == s[j].X {
		return s[i].Y < s[j].Y
	}
	return s[i].X < s[j].X
}

func (s Set) Swap(i, j int) {
	s[i], s[j] = s[j], s[i]
}

func NewPoint(x, y int) *Point {
	return &Point{
		X: x,
		Y: y,
	}
}

func NewEdge(vertex0, vertex1 *Vertex, bidirectional bool) *Edge {
	return &Edge{
		Vertex0:       vertex0,
		Vertex1:       vertex1,
		Bidirectional: bidirectional,
	}
}

func NewVertex(index int) *Vertex {
	return &Vertex{
		Index: index,
		Edges: make([]*Edge, 0),
	}
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	n := nextInt(scanner)
	points := make([]*Point, n)
	id3 := make(map[int]Set)
	for i := 0; i < n; i++ {
		x := nextInt(scanner)
		y := nextInt(scanner)
		point := NewPoint(x, y)
		points[i] = point
		if _, ok := id3[y]; !ok {
			id3[y] = make(Set, 0)
		}
		id3[y] = append(id3[y], point)
	}

	id49 := make(map[int]*Point)
	for _, set := range id3 {
		sort.Sort(set)
		lastX := set[0]
		for i := 1; i < len(set); i++ {
			point := set[i]
			NewEdge(point, lastX, true)
			if lastY, ok := id49[point.X]; ok {
				NewEdge(point, lastY, true)
			}
			id49[point.X] = point
			lastX = point
		}

File: output/GPT3.5-Turbo/xCodeEval_4000/Java/C/c60f78608a1a11e7f7f98789f3e9931c.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Point {
    int x;
    int y;
} Point;

typedef struct Edge {
    struct Point* vertex0;
    struct Point* vertex1;
    int bidirectional;
} Edge;

typedef struct Vertex {
    int index;
    struct Edge** edges;
    int num_edges;
} Vertex;

typedef struct Graph {
    struct Vertex** vertices;
    int num_vertices;
} Graph;

Graph* createGraph(int num_vertices) {
    Graph* graph = (Graph*)malloc(sizeof(Graph));
    graph->vertices = (Vertex**)malloc(num_vertices * sizeof(Vertex*));
    graph->num_vertices = num_vertices;
    for (int i = 0; i < num_vertices; i++) {
        graph->vertices[i] = (Vertex*)malloc(sizeof(Vertex));
        graph->vertices[i]->index = i;
        graph->vertices[i]->edges = NULL;
        graph->vertices[i]->num_edges = 0;
    }
    return graph;
}

void addEdge(Graph* graph, int vertex0, int vertex1, int bidirectional) {
